### DISCUSSION TITLE
is there any way to change the link vertices by a 90-degree angle

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2172] [Q&A]
### Introduction

![123](https://user-images.githubusercontent.com/117301232/236802305-5aa816c7-f05d-4f7a-bada-331658314cad.png)

when I am changing the vertices it should only allow the user to add vertices by 90 degrees. when we add vertices at different angles it should automatically change to 90  degree

in the first picture, I added vertices by around 45 degrees. 
I need to automatically change the angle by 90

![1233](https://user-images.githubusercontent.com/117301232/236802901-7fa6b100-36f1-4fd5-930a-ddfe6df87334.png)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
I assume you are looking for the [orthogonal router](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#routers.orthogonal).

**Reply by reshin-pulapaddy** [reply]
Thank you @kumilingus
it is working, after changing the router from normal to orthogonal it is not storing the vertices positions, is there a way to get all the vertices' point?
code : linkView.model.vertices()

**Reply by kumilingus** [reply]
What it is you are trying to achieve?

**Vertices** are user defined link way points. **Route** is an array of computed way points (based on the chosen `router` and input `vertices`).

**Reply by reshin-pulapaddy** [reply]
@kumilingus 
![Screenshot 2023-05-11 133716](https://github.com/clientIO/joint/assets/117301232/503f0040-aa07-4375-a8cc-9f9ceffc6dcc)
I need those 90-degree bend points

**Reply by kumilingus** [reply]
In general. Why is it important to have the vertices at the corners?

**Comment by reshin-pulapaddy** [comment]
![ezgif com-video-to-gif (1)](https://github.com/clientIO/joint/assets/117301232/de36f69f-841a-4b9b-b8c7-e7c9236653a2)

eg: if I am dropping a shape on top of link sources or targets it will automatically find the angle and rotate the shape connected to the end of the link

for finding the angle I am using vertices points,
after changing the router from normal to orthogonal it is not getting the vertices points (link bend points)
is there is any way to get computed way points (based on the chosen router and input vertices).

**Reply by kumilingus** [reply]
You can ask the link view instead of the model.

- [linkView.getConnection()](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#dia.LinkView.prototype.getConnection) returns a geometry path of the link
- [path.closestPointTangent()](https://resources.jointjs.com/docs/jointjs/v3.6/geometry.html#g.Path.prototype.closestPointTangent) returns the tangent at given point
- [tangent.angle()](https://resources.jointjs.com/docs/jointjs/v3.6/geometry.html#g.Line.prototype.angle) returns an angle of the tangent
- [g.intersection.exists(rect, path)](https://resources.jointjs.com/docs/jointjs/v3.6/geometry.html#g.intersection.exists) returns `true` if the rectangle (the shape) and the link intersects

**Reply by reshin-pulapaddy** [reply]
Thank you @kumilingus
